state-chain/runtime/src/chainflip.rs

```diff
@@ -1021,3 +1022,42 @@ impl cf_traits::MinimumDeposit for MinimumDepositProvider {
 		}
 	}
 }
+
+use sp_runtime::Saturating;
+
+pub struct CfTransactionFeeScaler;
+impl TransactionFeeScaler<RuntimeCall, AccountId, FlipBalance> for CfTransactionFeeScaler {
+	fn call_info(call: &RuntimeCall, caller: &AccountId) -> Option<CallInfoId<AccountId>> {
+		match call {
+			RuntimeCall::LiquidityPools(pallet_cf_pools::Call::set_limit_order {
+				base_asset,
+				..
+			}) |
+			RuntimeCall::LiquidityPools(pallet_cf_pools::Call::update_limit_order {
+				base_asset,
+				..
+			}) |
+			RuntimeCall::LiquidityPools(pallet_cf_pools::Call::set_range_order {
+				base_asset,
+				..
+			}) |
+			RuntimeCall::LiquidityPools(pallet_cf_pools::Call::update_range_order {
+				base_asset,
+				..
+			}) => Some(CallInfoId::Pool(PoolTouched {
+				account: caller.clone(),
+				base_asset: *base_asset,
+			})),
+			_ => None,
+		}
+	}
+
+	fn scale_fee(
+		pre_scaled_fee: FlipBalance,
+		scale_factor: u16,
+		exp_base: FixedU64,
+	) -> FlipBalance {
+		let multiplier = exp_base.saturating_pow(scale_factor.into());
+		multiplier.saturating_mul_int(pre_scaled_fee)
```

Comment:
I'm not sure this works as intended - if the scale factor is zero, the multiplier will be one. But we want to scale it down, right? Or am I misreading something?

Reply:
I think it's just the fact that this is implemented in terms of a `scale_factor` rather than, say, `call_count`?

Reply:
This was primarily to avoid needing to have an extra parameter on the trait, only to do the substraction inside. I had it inside before, and then took it out.